tests/test_isrTaskLSST.py

```diff
@@ -20,396 +20,801 @@
+        input_exp = mock.run()
+
+        isr_config = self.get_isr_config_electronic_corrections()
+        isr_config.doBias = True
+        isr_config.doDark = True
+        isr_config.doFlat = True
+        # We turn off defect masking to test the saturation code.
+        # However, the same pixels below should be masked/interpolated.
+        isr_config.doDefect = False
+
+        # This code will set the gain of one amp to the same as the ptc
+        # value, and we will check that it is logged and used but the
+        # results should be the same.
+        detectorConfig = isr_config.overscanCamera.getOverscanDetectorConfig(self.detector)
+        overscanAmpConfig = copy.copy(detectorConfig.defaultAmpConfig)
+        overscanAmpConfig.gain = self.ptc.gain[self.detector[1].getName()]
+        detectorConfig.ampRules[self.detector[1].getName()] = overscanAmpConfig
+
+        isr_task = IsrTaskLSST(config=isr_config)
+        with self.assertLogs(level=logging.WARNING) as cm:
+            result = isr_task.run(
+                input_exp.clone(),
+                bias=self.bias,
+                dark=self.dark,
+                flat=self.flat,
+                crosstalk=self.crosstalk,
+                defects=self.defects,
+                ptc=self.ptc,
+                linearizer=self.linearizer,
+            )
+        self.assertIn("Overriding gain", cm.output[0])
+
+        # Confirm that the output has the defect line as saturated.
+        sat_val = 2**result.exposure.mask.getMaskPlane("SAT")
+        for defect in self.defects:
+            np.testing.assert_array_equal(
+                result.exposure.mask[defect.getBBox()].array & sat_val,
+                sat_val,
+            )
+
+        clean_mock_config = self.get_mock_config_clean()
+        # We want the dark noise for more direct comparison.
+        clean_mock_config.doAddDarkNoiseOnly = True
+        clean_mock_config.doAddSky = True
+        clean_mock_config.doAddSource = True
+
+        clean_mock = isrMockLSST.IsrMockLSST(config=clean_mock_config)
+        clean_exp = clean_mock.run()
+
+        delta = result.exposure.image.array - clean_exp.image.array
+
+        bad_val = 2**result.exposure.mask.getMaskPlane("BAD")
+        good_pixels = np.where((result.exposure.mask.array & (sat_val | bad_val)) == 0)
+
+        # We compare the good pixels in the entirety.
+        self.assertLess(np.std(delta[good_pixels]), 5.0)
+        self.assertLess(np.max(np.abs(delta[good_pixels])), 5.0*5)
+
+        # Make sure the corrected image is overall consistent with the
+        # straight image.
+        self.assertLess(np.abs(np.median(delta[good_pixels])), 0.5)
+
+        # And overall where the interpolation is a bit worse but
+        # the statistics are still fine.  Note that this is worse than
+        # the defect case because of the widening of the saturation
+        # trail.
+        self.assertLess(np.std(delta), 7.0)
+
+    def get_mock_config_no_signal(self):
```

Comment:
Are these configurations things we might expect to be useful outside of the tests, and if so, should they be Mock classes?